web/src/js/utils.js

```diff
@@ -105,3 +105,17 @@ fetchApi.put = (url, json, options) => fetchApi(
         ...options
     }
 )
+
+
+export function getDiff(obj1, obj2) {
+    let result = {...obj2};
+    _.forIn(obj1, (value, key) => {
+    	  if(_.isEqual(obj2[key], obj1[key]))
+            result[key] = undefined;
+        else if(typeof Array.isArray(obj2[key]) && Array.isArray(obj2[key]))
+            result[key] = {...obj2[key]};
```

Comment:
I think arrays do not need this special case here at all. First, that assignment should have been `[...obj2[key]]`, not `{...obj2[key]}`. Second, `result = {...obj2};` already copies over the value as-is. As everything is immutable, we don't need to create a copy of the array.

I think this should clear up itself when writing tests :smiley: